commec/tools/database_handler.py

```diff
@@ -0,0 +1,142 @@
+        """ Virtual function to be called by any child database implementation"""
+        raise NotImplementedError(
+            "This class must override the .screen() to use DatabaseHandler."
+        )
+
+    def get_version_information(self) -> DatabaseVersion:
+        """ Override to call version retrieval information on a database."""
+        new_version_info = DatabaseVersion("", "", "Version retrieval not yet supported for this database")
+        return new_version_info
+
+    def check_output(self):
+        """ 
+        Simply checks for the existance of the output file, 
+        indicating that the database screening ran. Can be overrided if
+        more complex checks for a particular db is desired.
+        """
+        if os.path.isfile(self.out_file): # Consider adding checks from File_Tools such that empty outputs are invalid?
+            return True
+        return False
+
+    def __init__(self, database_file : str, input_file : str, out_file : str):
+        self.db_directory = os.path.dirname(database_file)
+        self.db_file = database_file
+        self.out_file = out_file
+        self.input_file = input_file
+        self.temp_log_file = f"{self.out_file}.log.tmp"
+        self.arguments_dictionary = {}
+        self.validate_directory()
+
+    # End override API
+    def check_input(self):
+        """ 
+        Simply checks for the existance of the input file, This is separated 
+        from database location, as sometimes the input file is generated at a 
+        previous step, and may not exist during DatabaseHandler instantiation.
+        """
+        if os.path.isfile(self.input_file):
+            return True
+        return False
+
+    @staticmethod
+    def is_empty(filepath: str) -> bool:
+        """Check if a file is empty or non-existent."""
+        try:
+            return os.path.getsize(os.path.abspath(os.path.expanduser(filepath))) == 0
+        except OSError:
+            # Errors such as FileNotFoundError considered empty
+            return True
+
+    @staticmethod
+    def has_hits(filepath: str) -> bool:
+        """Check if a file has any hits (lines that do not start with '#')."""
+        try:
+            with open(filepath, "r", encoding="utf-8") as file:
+                return any(not line.strip().startswith("#") for line in file)
+        except FileNotFoundError:
+            return False
+
+    def get_arguments(self) -> list:
+        """ 
+        convert the arguments dictionary into a list, 
+        structurally ready for appending to a command list of strs.
+        """
+        my_list = []
+        for key, value in self.arguments_dictionary.items():
+            my_list.append(str(key))
+            if isinstance(value, list):
+                my_list.append(" ".join(value))  # Extend the list with all elements in the array
+            else:
+                my_list.append(str(value))  # Append the value directly if it's not a list
+        return my_list
```

Comment:
Might make sense to implement this as an `@property` called `arguments_list` or `arguments_for_command`?

I think you'll also want to do  `my_list.append(" ".join(map(str, value)))` (i.e. explicitly case to string)